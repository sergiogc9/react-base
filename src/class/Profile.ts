export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
