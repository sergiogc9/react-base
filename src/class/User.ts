export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
