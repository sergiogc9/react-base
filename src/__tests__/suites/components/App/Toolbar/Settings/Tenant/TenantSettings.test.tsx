import { mount } from "enzyme";
import React from "react";

import { Settings as TenantSettingsObject } from "@src/class/Tenant";
import TenantSettings from "@src/components/App/Toolbar/Settings/Tenant/TenantSettings";
import TestHelper from "@src/__tests__/utils/Helper/TestHelper";

const tenantSettingsObject: TenantSettingsObject = {
	categorization_mode: "all",
	facebook_url: "https://www.facebook.com/conjunt.chapo",
	currency: "USD",
	display_influencers: true,
	valuation_metric: "miv",
	document_tenant_indices_search: false
};

const user = TestHelper.getUser();

const updateSettingsMock = jest.fn();
const updateFacebookUrlMock = jest.fn();
const closeModalMock = jest.fn();

describe("TenantSettings", () => {
	let wrapper: any;

	beforeEach(() => {
		wrapper = mount(
			<TenantSettings
				tenantSettings={tenantSettingsObject}
				loading={false}
				loadingFacebookImageUrl={false}
				facebookImageUrl=""
				onUpdateFacebookUrl={updateFacebookUrlMock}
				onUpdateTenantSettings={updateSettingsMock}
				onAddNotification={jest.fn()}
				onClose={closeModalMock}
				user={user}
			/>
		);
	});

	it("Test component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test update facebook url event handler", () => {
		wrapper.find('TextField#tenantFacebookUrl').simulate('focus');
		wrapper.find('SelectField#tenantCurrency').simulate('focus');
		expect(updateFacebookUrlMock).toHaveBeenCalledTimes(2); // One time on component mount and other after click events
	});

	it("Test clear facebook_url input", () => {
		expect(wrapper.state().settings.facebook_url).toStrictEqual('https://www.facebook.com/conjunt.chapo');
		wrapper.find('Button#clearFacebookUrlInput').simulate('click');
		expect(wrapper.state().settings.facebook_url).toStrictEqual('');
	});

	it("Test click save button event handler when no changes", () => {
		wrapper.find('Button#tenantSettingsSave').simulate('click');
		expect(updateSettingsMock).toHaveBeenCalledTimes(0);
	});

	it("Test click save button event handler when changes", () => {
		wrapper.find('Button#clearFacebookUrlInput').simulate('click');
		wrapper.find('Button#tenantSettingsSave').simulate('click');
		expect(updateSettingsMock).toHaveBeenCalledTimes(1);
	});

	it("Test update facebook image loading", () => {
		wrapper.setProps({
			...wrapper.props(),
			loadingFacebookImageUrl: true
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test close dialog button", () => {
		wrapper.find('Button#closeTenantSettingsButton').simulate('click');
		expect(closeModalMock).toHaveBeenCalled();
	});
});
