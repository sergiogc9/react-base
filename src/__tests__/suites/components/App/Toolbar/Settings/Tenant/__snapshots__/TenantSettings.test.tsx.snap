// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenantSettings Test component snapshot 1`] = `"<div class=\\"tenant-settings\\"><div class=\\"tenant-settings-header md-grid\\"><i class=\\"md-icon material-icons\\">person</i><span>tenant.settings.header</span><button id=\\"closeTenantSettingsButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block close-btn btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button></div><div class=\\"tenant-settings-form md-grid md-grid--stacked\\"><div><form><div class=\\"tenant-settings-form-field\\"><div class=\\"md-text-field-container md-inline-block md-text-field-container--input\\"><label for=\\"tenantFacebookUrl\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">tenant.settings.facebook_page.label</label><input id=\\"tenantFacebookUrl\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-text\\" value=\\"https://www.facebook.com/conjunt.chapo\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div><button id=\\"clearFacebookUrlInput\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button><button id=\\"helpFacebookUrlInput\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-btn--tooltip md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-tooltip-container\\"><span aria-hidden=\\"true\\"></span></div><i class=\\"md-icon material-icons md-text--inherit\\">help_outline</i></button></div><div></div></form></div><div class=\\"tenant-settings-form-field\\"><div id=\\"tenantCurrency-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field tenant-settings-select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"tenantCurrency\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">tenant.settings.currency</label><div id=\\"tenantCurrency-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">USD ($)</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"tenantCurrency\\" value=\\"USD\\"></div></div></div></div><div class=\\"tenant-settings-form-field\\"><div id=\\"tenantAvailableCategories-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field tenant-settings-select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"tenantAvailableCategories\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">tenant.settings.categorization_mode.value</label><div id=\\"tenantAvailableCategories-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">tenant.settings.categorization_mode.all</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"tenantAvailableCategories\\" value=\\"all\\"></div></div></div></div><div class=\\"tenant-settings-form-buttons md-grid\\"><button id=\\"tenantSettingsCancel\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--red-ink\\"><div class=\\"md-ink-container\\"></div>filters.period.custom.dialog.cancel</button><button id=\\"tenantSettingsSave\\" type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">filters.period.custom.dialog.ok</span></div></button></div></div></div>"`;

exports[`TenantSettings Test update facebook image loading 1`] = `"<div class=\\"tenant-settings\\"><div class=\\"tenant-settings-header md-grid\\"><i class=\\"md-icon material-icons\\">person</i><span>tenant.settings.header</span><button id=\\"closeTenantSettingsButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block close-btn btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button></div><div class=\\"tenant-settings-form md-grid md-grid--stacked\\"><div><form><div class=\\"tenant-settings-form-field\\"><div class=\\"md-text-field-container md-inline-block md-text-field-container--input\\"><label for=\\"tenantFacebookUrl\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">tenant.settings.facebook_page.label</label><input id=\\"tenantFacebookUrl\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-text\\" value=\\"https://www.facebook.com/conjunt.chapo\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div><button id=\\"clearFacebookUrlInput\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button><button id=\\"helpFacebookUrlInput\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-btn--tooltip md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-tooltip-container\\"><span aria-hidden=\\"true\\"></span></div><i class=\\"md-icon material-icons md-text--inherit\\">help_outline</i></button></div><div><svg id=\\"tenantSettingsFacebookImageLoading\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-progress--circular-indeterminate\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div></form></div><div class=\\"tenant-settings-form-field\\"><div id=\\"tenantCurrency-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field tenant-settings-select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"tenantCurrency\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">tenant.settings.currency</label><div id=\\"tenantCurrency-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">USD ($)</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"tenantCurrency\\" value=\\"USD\\"></div></div></div></div><div class=\\"tenant-settings-form-field\\"><div id=\\"tenantAvailableCategories-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field tenant-settings-select-field\\"><div class=\\"md-select-field__toggle\\"><label for=\\"tenantAvailableCategories\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">tenant.settings.categorization_mode.value</label><div id=\\"tenantAvailableCategories-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">tenant.settings.categorization_mode.all</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"tenantAvailableCategories\\" value=\\"all\\"></div></div></div></div><div class=\\"tenant-settings-form-buttons md-grid\\"><button id=\\"tenantSettingsCancel\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--red-ink\\"><div class=\\"md-ink-container\\"></div>filters.period.custom.dialog.cancel</button><button id=\\"tenantSettingsSave\\" type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">filters.period.custom.dialog.ok</span></div></button></div></div></div>"`;
